frontstage/views/surveys/add_survey_submit.py

```diff
@@ -31,15 +31,25 @@ def add_survey_submit(session):
         case_id = iac['caseId']
         case = case_controller.get_case_by_enrolment_code(enrolment_code)
         business_party_id = case['partyId']
-        case_controller.post_case_event(case_id,
-                                        party_id=business_party_id,
-                                        category='ACCESS_CODE_AUTHENTICATION_ATTEMPT',
-                                        description='Access code authentication attempted')
+        collection_exercise_id = case['caseGroup']['collectionExerciseId']
+        # Get survey ID from collection Exercise
+        added_survey_id = collection_exercise_controller.get_collection_exercise(
+            case['caseGroup']['collectionExerciseId']).get('surveyId')
 
-        party_controller.add_survey(party_id, enrolment_code)
+        info = party_controller.get_party_by_business_id(business_party_id, collection_exercise_id)
 
-        # Get survey ID from collection Exercise
-        added_survey_id = collection_exercise_controller.get_collection_exercise(case['caseGroup']['collectionExerciseId']).get('surveyId')
+        already_enrolled = None
+        if is_business_enrolled(info['associations'], case['caseGroup']['surveyId']):
+            logger.error('User tried to enrol onto a survey they are already enrolled on',
```

Comment:
Really, this isn't an error or even a warning - it's the user's fault for doing this. Errors and warnings should be for things where the system isn't working properly. I would consider doing this as info level logging.

Reply:
Changed